sdks/python/apache_beam/dataframe/io.py

```diff
@@ -540,10 +540,15 @@ def process(
     reader = self.reader
     if isinstance(reader, str):
       reader = getattr(pd, self.reader)
+    indices_per_file = 10**int(math.log(2**64 // len(path_indices), 10))
+    if readable_file.metadata.size_in_bytes > indices_per_file:
+      raise RuntimeError(
+          f'Cannot safely index records from {len(path_indices)} files '
+          f'of size {readable_file.metadata.size_in_bytes} '
+          f'as their product is greater than 2^63.')
```

Comment:
This is a little conservative, couldn't we detect this based on the number of rows, rather than using byte size as a proxy?

Reply:
It's harder to know the number of rows ahead of time (though we could keep track after the fact), but more importantly when we split we start the index based on the byte offset so we can't actually use most of that headroom anyway.

Reply:
ah yeah I didn't make the connection that `start_index` is mixing bytes and elements, thanks for clarifying.